File: Ruby/uri-1036.rb
# frozen_string_literal: true

# URI - 1036: https://www.urionlinejudge.com.br/judge/pt/problems/view/1036

def calculate_bhaskara_roots(values)
  delta = values[1]**2 - (4 * values[0] * values[2])
  divider = 2 * values[0]
  return 'Impossivel calcular' if delta.negative? || divider.zero?

  { R1: ((-values[1] + Math.sqrt(delta)) / divider).round(5),
    R2: ((-values[1] - Math.sqrt(delta)) / divider).round(5) }
end

values = gets.chomp.split(' ').map(&:to_f)
result = calculate_bhaskara_roots(values)
if result.is_a?(String)
  puts result
else
  puts "R1 = #{result[:R1]}"
  puts "R2 = #{result[:R2]}"
end

File: URI/C++/uri-1036.cpp
// URI 1036: https://www.urionlinejudge.com.br/judge/en/problems/view/1036

#include<iostream>
#include<iomanip>
#include <cmath>

using namespace std;

int main() {
  double a, b, c;
  cin >> a >> b >> c;
  double delta = (b*b) - 4*a*c;
  if (delta < 0 || 2*a == 0) {
    cout << "Impossivel calcular" << endl;
  } else {
    cout.precision(5);
    cout << "R1 = "  << fixed << (-b + (sqrt(delta))) / (2*a) << endl;
    cout << "R2 = "  << fixed << (-b - (sqrt(delta))) / (2*a) << endl;
  }

  return 0;
}
